src/Query/QueryBuilder.php

```diff
@@ -300,19 +300,57 @@ public function fetchFirstColumn(): array
         return $this->connection->fetchFirstColumn($this->getSQL(), $this->params, $this->paramTypes);
     }
 
+    /**
+     * Executes an SQL query (SELECT) and returns a Result.
+     *
+     * @throws Exception
+     */
+    public function executeQuery(): Result
+    {
+        return $this->connection->executeQuery($this->getSQL(), $this->params, $this->paramTypes);
+    }
+
+    /**
+     * Executes an SQL statement and returns the number of affected rows.
+     *
+     * Should be used for INSERT, UPDATE and DELETE
```

Comment:
It's not limited to DML (`INSERT`, etc.), it could be also DDL and other non-DQL statements. See [JDBC](https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html) for reference.

Reply:
Idem, the `QueryBuilder` cannot build anything else than DML. Should we really mention that ?

Reply:
No, you're right.